src/repositories/customer.ts

```
@@ -85,5 +87,19 @@ export class CustomerRepository extends AbstractResourceRepository<'customer'> {
         400
       )
     },
+    setCustomField: (
+      _context: RepositoryContext,
+      resource: Writable<Customer>,
+      { name, value }: CustomerSetCustomFieldAction
+    ) => {
+      if (!resource.custom) {
```

If the resource has no custom type defined it should throw an error, see the other resources as example

---

Fixed, please check if the error message is acceptable.